dask/dataframe/indexing.py

```diff
@@ -36,6 +36,9 @@ def _make_meta(self, iindexer, cindexer):
         else:
             return self._meta_indexer[:, cindexer]
 
+    def __dask_tokenize__(self):
+        return type(self).__name__, tokenize(self.obj)
```

Comment:
```suggestion
        return type(self).__name__, self.obj
```

Would be sufficient, because tokenized gets mapped across the tuple :)

Reply:
Unfortunately, it's not sufficient, look at the following example:

```python
def test_deterministic_hashing_dataframe_():
    # Add duplicate column names in order to use _iLocIndexer._iloc path
    obj = pd.DataFrame([[0, 1, 2, 3], [4, 5, 6, 7]], columns=["a", "b", "c", "c"])

    dask_df = dd.from_pandas(obj, npartitions=1)
    dask_df_double = dd.from_pandas(obj * 2, npartitions=1)

    ddf1 = dask_df.loc[0:1, ["a", "c"]]
    ddf2 = dask_df_double.loc[0:1, ["a", "c"]]

    assert tokenize(ddf1) == tokenize(ddf2)
```

Without token(self.obj), this test will pass even if the two dask dataframes are not reading the same data.
It's due to the implementation of tokenize:

```python
def tokenize(*args, **kwargs):
    """Deterministic token

    >>> tokenize([1, 2, '3'])
    '7d6a880cd9ec03506eee6973ff551339'

    >>> tokenize('Hello') == tokenize('Hello')
    True
    """
    hasher = _md5(str(tuple(map(normalize_token, args))).encode())
    if kwargs:
        hasher.update(str(normalize_token(kwargs)).encode())
    return hasher.hexdigest()
```

str(dask_df ) and str(dask_df_double) returns the same string so the hash will be the same for dask data frame with the same shape, number of partitions, and built from the same function...

Reply:
Ah, you're correct -- I thought we were `map`ping `normalize_token` across the values returned by `__dask_tokenize__` but that's not the case